sgkit/stats/grm.py

```diff
@@ -147,3 +148,522 @@ def genomic_relationship(
+    ValueError
+        If ``genomic_sample_index`` is not specified and the genomic and pedigree
+        relationship matrices do not have the same dimensions.
+    ValueError
+        If the genomic relationship matrix is larger than the pedigree
+        relationship matrix.
+    ValueError
+        If ``pedigree_subset_inverse_relationship`` is specified and its dimensions
+        do not match those of the genomic relationship matrix.
+
+    Examples
+    --------
+
+    >>> import xarray as xr
+    >>> import sgkit as sg
+    >>> ds = xr.Dataset()
+    >>> # A-matrix
+    >>> ds["pedigree_relationship"] = ["samples_0", "samples_1"], [
+    ...     [1.   , 0.   , 0.   , 0.5  , 0.   , 0.25 ],
+    ...     [0.   , 1.   , 0.   , 0.5  , 0.5  , 0.5  ],
+    ...     [0.   , 0.   , 1.   , 0.   , 0.5  , 0.25 ],
+    ...     [0.5  , 0.5  , 0.   , 1.   , 0.25 , 0.625],
+    ...     [0.   , 0.5  , 0.5  , 0.25 , 1.   , 0.625],
+    ...     [0.25 , 0.5  , 0.25 , 0.625, 0.625, 1.125]
+    ... ]
+    >>> # G-matrix
+    >>> ds["genomic_relationship"] = ["genotypes_0", "genotypes_1"], [
+    ...     [1.3 , 0.3 , 0.71],
+    ...     [0.3 , 1.3 , 0.73],
+    ...     [0.71, 0.73, 1.45],
+    ... ]
+    >>> # indices of G-matrix samples within A-matrix
+    >>> ds["genomic_sample_index"] = "genotypes", [3, 4, 5]
+    >>> ds = sg.hybrid_relationship(
+    ...     ds,
+    ...     pedigree_relationship="pedigree_relationship",
+    ...     genomic_relationship="genomic_relationship",
+    ...     genomic_sample_index="genomic_sample_index",
+    ... )
+    >>> ds.stat_hybrid_relationship.values.round(3)  # doctest: +NORMALIZE_WHITESPACE
+    array([[ 1.084,  0.056, -0.028,  0.653, -0.013,  0.281],
+           [ 0.056,  1.112,  0.056,  0.64 ,  0.64 ,  0.576],
+           [-0.028,  0.056,  1.084, -0.013,  0.653,  0.295],
+           [ 0.653,  0.64 , -0.013,  1.3  ,  0.3  ,  0.71 ],
+           [-0.013,  0.64 ,  0.653,  0.3  ,  1.3  ,  0.73 ],
+           [ 0.281,  0.576,  0.295,  0.71 ,  0.73 ,  1.45 ]])
+
+    References
+    ----------
+    [1] - J. W. R. Martini, M. F. Schrauf, C. A. Garcia-Baccino, E. C. G. Pimentel, S. Munilla,
+    A. Roberg-Muñoz, R. J. C. Cantet, C. Reimer, N. Gao, V. Wimmer and H. Simianer 2018.
+    "The effect of the :math:`H^{-1}` scaling factors :math:`\\tau` and :math:`\\omega`
+    on the structure of :math:`H` in the single-step procedure."
+    Genetics Selection Evolution 50 (16).
+
+    [2] - A. Legarra, I. Aguilar and I. Misztal 2009.
+    "A relationship matrix including full pedigree and genomic information."
+    Journal of Dairy Science 92 (9): 4656-4663.
+
+    [3] - I. Aguilar, I. Misztal, D. L. Johnson, A. Legarra, S. Tsuruta and T. J. Lawlor 2010.
+    "A unified approach to utilize phenotypic, full pedigree, and genomic information for genetic
+    evaluation of Holstein final score."
+    Journal of Dairy Science 93 (2): 743-752.
+
+    """
+    estimator = estimator or "Martini"
```

Comment:
Best to be explicit here so that arguments such as `False` raise the right error.
```suggestion
    if estimator is None:
         estimator = "Martini"
```